tests/IotServiceTest.cpp

```diff
@@ -7,26 +7,54 @@
 
 #include <aws/testing/aws_test_harness.h>
 
+#include <aws/common/environment.h>
+#include <aws/common/string.h>
+#include <aws/crt/UUID.h>
+
 #include <condition_variable>
 #include <fstream>
 #include <mutex>
 #include <utility>
 
 #include <aws/io/logging.h>
 
-#define TEST_CERTIFICATE "/tmp/certificate.pem"
-#define TEST_PRIVATEKEY "/tmp/privatekey.pem"
-#define TEST_ROOTCA "/tmp/AmazonRootCA1.pem"
-
 #if !BYO_CRYPTO
 
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt311_test_envName_iot_core_host, "AWS_TEST_MQTT311_IOT_CORE_HOST");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt311_test_envName_iot_core_cert, "AWS_TEST_MQTT311_IOT_CORE_RSA_CERT");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt311_test_envName_iot_core_key, "AWS_TEST_MQTT311_IOT_CORE_RSA_KEY");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt311_test_envName_iot_core_ca, "AWS_TEST_MQTT311_ROOT_CA");
+
 static int s_TestIotPublishSubscribe(Aws::Crt::Allocator *allocator, void *ctx)
 {
     using namespace Aws::Crt;
     using namespace Aws::Crt::Io;
     using namespace Aws::Crt::Mqtt;
 
-    const char *credentialFiles[] = {TEST_CERTIFICATE, TEST_PRIVATEKEY, TEST_ROOTCA};
+    // ========== Environment variable setup
+    aws_string *input_host;
```

Comment:
Trivial: The Environment variable setup of aws_strings, envResult, and isEndValid is repeated in multiple tests. Could we set this up as its own singular struct that can be used across all the tests? the isEnvValid could also be set to be false if any of the envResults don't return an AWS_OP_SUCCESS so we don't need to expose an additional int envResult. That would also allow the skip test cleanup to be done in a call to a function with the collection of strings rather than repeating the string destroy block of code. Again, very trivial.

Reply:
We could alternately have a function that takes the 4 strings and returns a bool on whether they've all been successfully set or not. On fail they can be cleaned up in the function and you can immediately skip. Or we can just leave it as is since it's perfectly fine and I'm just overthinking this >,<